src/diagnostics/MOM_diagnose_MLD.F90

```diff
@@ -362,9 +395,9 @@ subroutine diagnoseMLDbyEnergy(id_MLD, h, tv, G, GV, US, Mixing_Energy, diagPtr,
         H_ML_TST = 0.0
         PE_Mixed = 0.0
 
-        do k=1,nz
+        do k=k_bounds(1),k_bounds(2),k_int
 
-          ! This is the unmixed PE cumulative sum from top down
+          ! This is the unmixed PE cumulative sum in the direction k_int
           PE = PE + 0.5 * Rho_c(i,k) * (Z_int(K)**2 - Z_int(K+1)**2)
```

Comment:
A more versions of this line would be `PE = PE + 0.5 * Rho_c(i,k) * dZ(i,k) * (2.*Z_int(K) - dZ(i,k)**2)` or equivalently (with no subtraction) `PE = PE + 0.5 * Rho_c(i,k) * dZ(i,k) * (Z_int(K) + Z_int(K+1))`.  However, this would change this diagnostic at roundoff, so maybe we should not do this.

Reply:
I can make this change (with the use_OM4_iteration flag), but it adds an extra "if" within the loop.  We can discuss how necessary it is.  We could also note the suggestion as a comment in the code

Reply:
Added the flag as suggested.